tests/upgrades/test_host.py

```diff
@@ -0,0 +1,154 @@
+"""Test Host/Provisioning related Upgrade Scenario's
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Hosts
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import random
+
+from airgun.session import Session
+from fauxfactory import gen_string
+from nailgun import entities
+from robottelo.api.utils import skip_yum_update_during_provisioning
+from robottelo.constants import FOREMAN_PROVIDERS, LATEST_RHEL7_GCE_IMG_UUID, VALID_GCE_ZONES
+from robottelo.helpers import download_gce_cert
+from robottelo.test import APITestCase, settings
+from upgrade_tests import pre_upgrade, post_upgrade
+from upgrade_tests.helpers.scenarios import create_dict, get_entity_data
+from wait_for import wait_for
+
+
+GCE_SETTINGS = dict(
+    project_id=settings.gce.project_id,
+    client_email=settings.gce.client_email,
+    cert_path=settings.gce.cert_path,
+    zone=settings.gce.zone,
+    cert_url=settings.gce.cert_url,
+)
+
+
+class scenario_positive_gce_host_compute_resource(APITestCase):
+    """The host can be provisioned on GCE CR created in previous version
+
+    :steps:
+
+        1. In Preupgrade Satellite, create GCE Compute Resource
+        2. Upgrade the satellite to next/latest version
+        3. Postupgrade, The Compute Resource attributes can be manipulated
+        4. The host can be provisioned on GCE CR created in previous satellite version
+
+    :expectedresults:
+
+        1. The host should be provisioned on GCE CR created in previous version
+        2. The GCE CR attributes should be manipulated
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.fullhost = None
+        cls.arch = entities.Architecture().search(query={'search': 'name=x86_64'})[0]
+        cls.os = entities.OperatingSystem().search(
+            query={'search': 'family=Redhat and major=7'})[0]
+        download_gce_cert(GCE_SETTINGS['cert_url'], GCE_SETTINGS['cert_path'])
+        cls.domain_name = settings.server.hostname.split('.', 1)[1]
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.fullhost:
+            hst = entities.Host().search(
+                query={'search': 'name={}'.format(cls.fullhost)})
+            if hst:
+                entities.Host(id=hst[0].id).delete()
+
+    @pre_upgrade
+    def test_pre_create_gce_cr_and_host(self):
+        """
+        """
+        cr_name = gen_string('alpha')
+        org = entities.Organization().create()
+        loc = entities.Location().create()
+        with Session('gce_upgrade_tests') as session:
```

Comment:
Any specific reason for creating GCE resource using AIrgun? Is that not possible with API?

Reply:
This is a pre part of upgrade scenario, that means its going to be run in 6.6. And 6.6 only supports UI not API.

Reply:
okay make sense, I was not aware that